src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/services/citizen/SettlementReachedCitizenNotificationServiceTest.java

```diff
@@ -0,0 +1,174 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static uk.gov.hmcts.cmc.claimstore.controllers.PathPatterns.CLAIM_REFERENCE_PATTERN;
+
+public class SettlementReachedCitizenNotificationServiceTest extends BaseOfferTest {
+
+    private static final String CLAIM_REFERENCE_NUMBER = CLAIM_REFERENCE_PATTERN.replace("^", "");
+    private static final String FRONTEND_BASE_URL = "https://civil-money-claims.co.uk";
+    private static final String OFFER_COUNTER_SIGNED_EMAIL_BY_ORIGINATOR = "9d1ddac9-d6a7-41f3-bfd4-dcfbcb61dcf1";
+    private static final String OFFER_COUNTERSIGNED_EMAIL_BY_OTHER_PARTY = "cfde3889-e202-4d70-bc64-f54048616be3";
+    private static final String COUNTER_SIGNING_PARTY_KEY = "counterSigningParty";
+    private static final String FRONTEND_BASE_URL_KEY = "frontendBaseUrl";
+
+    @Captor
+    private ArgumentCaptor<String> senderArgument;
+
+    @Captor
+    private ArgumentCaptor<String> templateArgument;
+
+    @Captor
+    private ArgumentCaptor<String> referenceArgument;
+
+    @Captor
+    private ArgumentCaptor<Map<String, String>> emailDataArgument;
+
+    @Test
+    public void verifyNotificationsWhenCounterSignedByDefendant() throws Exception {
+        given(pdfServiceClient.generateFromHtml(any(byte[].class), anyMap()))
+            .willReturn(PDF_CONTENT);
+
+        claimantRequestFor("accept");
+        defendantRequestFor("countersign");
+
+        verify(notificationClient, atLeast(6))
+            .sendEmail(templateArgument.capture(), senderArgument.capture(), emailDataArgument.capture(),
+                referenceArgument.capture());
+
+        verifyConfirmationToDefendantOnDefendantCounterSign();
+        verifyNotificationToClaimantOnDefendantCounterSign();
+
+    }
+
+    private void verifyConfirmationToDefendantOnDefendantCounterSign() {
+        assertThat(templateArgument.getValue()).isEqualTo(OFFER_COUNTER_SIGNED_EMAIL_BY_ORIGINATOR);
+
+        assertThat(referenceArgument.getValue())
+            .matches("to-defendant-agreement-counter-signed-by-defendant-notification-"
+                + CLAIM_REFERENCE_NUMBER
+            );
+
+        Map<String, String> emailData = new HashMap<>();
+        emailData.put(COUNTER_SIGNING_PARTY_KEY, claim.getClaimData().getDefendant().getName());
+        emailData.put(FRONTEND_BASE_URL_KEY, FRONTEND_BASE_URL);
+
+        verifyEmailData(emailDataArgument.getValue(), emailData);
+
+        assertThat(senderArgument.getValue())
+            .isEqualTo(claim.getClaimData().getDefendant().getEmail().orElse(null));
```

Comment:
Event handler does not use `claim.getClaimData().getDefendant().getEmail()`, it uses `claim.getDefendantEmail()`